### ISSUE TITLE
groups: cannot find name for group ID

### ISSUE THREAD

**Issue by hariswhdn** [Neo-Oli/termux-ubuntu#2]
$ ./startubuntu.sh
groups: cannot find name for group ID 1015
groups: cannot find name for group ID 1023
groups: cannot find name for group ID 1028
groups: cannot find name for group ID 3003
groups: cannot find name for group ID 9997
groups: cannot find name for group ID 50020
root@localhost:~# exit
logout

I get error while extracting file with tar, there was 6 file not extracted. I start ubuntu and get that error. So i extract ubuntu.tar.gz again with tar command, i get the 6 file back but the error still appear, is this not related with that error? I am new on linux, thank you.

**Comment by Neo-Oli** [comment]
The 6 or more errors when extracting the tar are actually normal. It's all files in ubuntu-fs/dev/, which gets mounted to /dev/ anyway, so it shouldn't matter. 

I also get the `groups:` errors. So far I don't know how to avoid them

**Comment by remo7777** [comment]
When decompress ubuntu**tar.gz shows decompressing ubuntu image
tar: dev/full: Cannot mknod: Operation not permitted
tar: dev/random: Cannot mknod: Operation not permitted
tar: dev/tty: Cannot mknod: Operation not permitted
tar: dev/urandom: Cannot mknod: Operation not permitted
tar: dev/null: Cannot mknod: Operation not permitted
tar: dev/zero: Cannot mknod: Operation not permitted
tar: Exiting with failure status due to previous errors
fixing nameserver, otherwise it can't connect to the internet

**Comment by remo7777** [comment]
When i lunch start.sh shows -- bash start.sh
groups: cannot find name for group ID 3003
groups: cannot find name for group ID 9997
groups: cannot find name for group ID 50137
mesg: change /dev/pts/1 mode failed: Permission denied

My dpkg architecture is aarch64 plz...help..

**Comment by runskey2017** [comment]
./start.sh
groups: cannot find name for group ID 1015
groups: cannot find name for group ID 1028
groups: cannot find name for group ID 3003
groups: cannot find name for group ID 9997
groups: cannot find name for group ID 50307
mesg: change /dev/pts/17 mode failed: Permission denied  
My dpkg is arm pls help

**Comment by maidenone** [comment]
When i lunch start.sh shows -- bash start.sh
groups: cannot find name for group ID 3003
groups: cannot find name for group ID 9997
groups: cannot find name for group ID 50137
mesg: change /dev/pts/1 mode failed: Permission denied

My dpkg architecture is aarch64

This is what i have as well

**Comment by Neo-Oli** [comment]
You can try this branch https://github.com/Neo-Oli/termux-ubuntu/tree/stubs to see if it fixes things for you without breaking other things.

**Comment by Ahmedroid** [comment]
why i cant go to sdacrd in ubunto
root@localhost:~# cd /sdcrad
bash: cd: /sdcrad: No such file or directory

**Comment by shaqman** [comment]
@Ahmedroid 

First of all, even though there is a /sdcard folder, which there isn't, you mistyped it there.
See termux-storage for mounting sdcard into termux, then uncomment inside start.sh
#command+=" -b /data/data/com.termux/files/home"
so it could be bind inside ubuntu.

Please note, that this is not a support forum. And you are hijacking a totally different issue.

**Comment by NatoBoram** [comment]
Same issue!
```bash
$ ./start-ubuntu.sh
groups: cannot find name for group ID 3003
groups: cannot find name for group ID 9997
groups: cannot find name for group ID 20167
groups: cannot find name for group ID 50167
root@localhost:~# apt
apt 1.6.1 (arm64)
```

**Comment by Root-voidX** [comment]
I have the same issue to:

groups: cannot find name for group ID 3003
groups: cannot find name for group ID 9997
groups: cannot find name for group ID 20290
groups: cannot find name for group ID 50290

**Comment by Ahmedroid** [comment]
I know that but the problem did not solved

**Comment by MFDGaming** [comment]
i recomend to you to use this https://play.google.com/store/apps/details?id=com.gnuroot.debian and not termux

**Comment by corbinlc** [comment]
This is an issue that must be solved with proot.  If the getgroups system call is called the kernel is returning supplemental group ids that your ubuntu chroot knowns nothing about.  Probably need to fully emulate getgroups and setgroups withing fake_id0 extension.  Right now there is a /*TODO*/ comment there.  As a shorter term fix, you can probably get proot to drop those system calls.

**Comment by corbinlc** [comment]
Yes.  Just dropping the system calls seems to not cause and issue for me.  My version of fake_id0 is very different than yours, but you get the point.  If I fully emulate this later, I will make sure to update you all.

**Comment by gauloics** [comment]
solved by
`touch ~/.hushlogin`

**Comment by diehard88** [comment]
touché!

**Comment by Amooti73** [comment]
Thanks, gauloics! It worked for me!

Br. John 

* gauloics <notifications@github.com> [2018-10-18 11:00:53 -0700]:

>solved by
>`touch ~/.hushlogin`
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/Neo-Oli/termux-ubuntu/issues/2#issuecomment-431105144

-- 
Bro. John Flood, C.S.C.

Email: jfloodcsc@gmail.com
jflood@cuea.edu
Phone: +254-711-983025/0711-983025
+256-779-427935

Jesus "expects us to stop looking for those personal or communal niches which shelter us from the maelstrom of human misfortune, and instead to enter into the reality of other people's lives and to know the power of tenderness.  Whenever we do so, our lives become wonderfully complicated".

Pope Francis, The Joy of the Gospel, 44

**Comment by AlbertCaro** [comment]
@gauloics I love u

**Comment by JacobTDC** [comment]
I'm not sure disabling the check is the best approach. I remember finding a way to fix it, by changing the startup file, but I don't remember. I'll leave my solution here if I find out.

**Comment by JacobTDC** [comment]
Ah, yes, I remember. Maybe using addgroup? Can anyone think of any potential problems with doing that?

**Comment by Neboer** [comment]
@gauloics I love u too! The way to solve the problem is to touch a file in your Ubuntu system and you cannot get error again!

**Comment by apolbox** [comment]
If you want, i make repo for install termux-ubuntu and how to fix groups: cannot find name group id

click this link https://github.com/apolbox/termux-ubuntu

**Comment by Nebdir** [comment]
after quick research the groups are found in the rootfs bc of android adding them. so if your in your termux shell you should see those missing ids by running `id`
add these groups to your groups in your chroot. you might have to add any none root user to a set of them in order to make sudo or some different programms to work.

**Comment by akashisgreat** [comment]
> why i cant go to sdacrd in ubunto
> root@localhost:~# cd /sdcrad
> bash: cd: /sdcrad: No such file or directory
Remove the `#` in `start-ubuntu.sh` , line no 58<br> 
 <a href="https://github.com/Neo-Oli/termux-ubuntu/blob/87e5818cedbfafff1e3fb42e751fc9a25671a48e/ubuntu.sh#L58">See here</a>
https://github.com/Neo-Oli/termux-ubuntu/blob/87e5818cedbfafff1e3fb42e751fc9a25671a48e/ubuntu.sh#L58
After that run `./start-ubuntu.sh` agian

**Comment by duskygloom** [comment]
Yesterday I also faced this issue, but there's no need to hushlogin to fix it. In my case, the problem arose because mounted the termux home folder.
`#command+=" -b /data/data/com.termux/files/home:/root"`
This changed the home directory of the distro to the home directory of termux.

To fix this, you may copy all the data from the distro's home to termux home.
Or you may recomment the line in the start-<distro>.sh

Note that this was the problem in my case. And it worked. If this is not what happened with you, you may add a blank file named .hushlogin in the termux home by
`touch .hushlogin`

This will reset all your customisations, but you will have all your applications intact.

**Comment by mashirozx** [comment]
In your termux run `groups`, you'll get this:
![image](https://user-images.githubusercontent.com/16148054/106879220-d4607f00-6715-11eb-8c98-5be181367d59.png)

And then `id GROUP_NAME_YOU_GOT`, you'll get something like this:
![image](https://user-images.githubusercontent.com/16148054/106880120-f5759f80-6716-11eb-9732-a165a10aec71.png)

You see, some group cannot show here, you need to check them by `id GID_NEED_TO_QUERY`.

This uid and the group name is what we need, in my condition, run commands below in Ubuntu:

```bash
addgroup --system --gid 3003 inet
addgroup --system --gid 9997 everybody
addgroup --system --gid 20237 u0_a237_cache
addgroup --system --gid 50237 all_a237
addgroup --system --gid 99909997 u999_everybody
reset
```
